### DISCUSSION TITLE
Zones/regions

### DISCUSSION THREAD

**Discussion by richard-fine** [OpenRCT2/OpenRCT2#21799] [Feature requests]
I'm sure this has been suggested before, but what do the dev team think of adding 'regions'/'zones'/'neighbourhoods' to the game? You see this often in real-world parks, like how Disney Anaheim has 'Frontierland', 'Fantasyland', 'Mickey's Toontown' etc all within one park.

As I imagine it: you'd define a region the same way you lay down patrol areas for staff, and once you define a region you'd be able to use it in a few ways:

- You'd be able to assign a staff member to a region (or maybe multiple regions), which would cause them to take their patrol area from the region definition instead of needing to draw it out for each staff member individually. If you change the region definition later (e.g. expanding it), all staff assigned to that region automatically patrol the new area.
- You'd be able to filter information by region, so that you can e.g. see thoughts from guests who are in a particular region as opposed to only the entire park, or see the info in the Rides window filtered to a single region so you can look at the popularity/profit/etc of things in that region only. For a ride to be considered part of a region, it must have at least one entrance on a tile which is assigned to the region.
- You'd be able to pull up a window for the region, which would let you do things like edit the region definition, name, and delete it, as well as view aggregated information about the region like the total number of guests in the region right now, the staff assigned to it, the total hourly income/profit, the list of attractions in the region, etc. There would also be a new tab in the Park Information window showing all the regions and giving you the button for creating a new one.
- Alerts (like 'Guests are complaining about the disgusting state of paths in your park') could change to report specific regions, rather than just 'the park'.

Other than that I don't imagine it having any impact on the simulation itself - it'd just be a management tool for dealing with larger parks.

WDYT?

**Comment by Basssiiie** [comment]
I like the idea, but I think it may be better suited as a plugin instead of a core game feature. :)

**Reply by richard-fine** [reply]
Is it actually possible to do it as a plugin? I assumed it'd need quite tight integration with things like the code that generates alerts.

**Reply by ltsSmitty** [reply]
I agree that a plugin would be a great path forward for this. Not everything would be able to be displayed in the native menus, but many things could be rebuilt in some custom windows. I'm interested in building it, but I need to do some more planning first. I'll reach out about some expected behavior that I'm still thinking about.

**Reply by dbophxlip2017** [reply]
What about using the Banners?  Maybe you could like the banners to a zone.  I wish there was more options for text tho, like having a couple of normal displays that could display after each other, and a couple for when its closed, like use the other exit, please exit the zone, or let us custom up to like 3 for open, 3 for closed.  Maybe a yellow status with "Coming Soon" The could then be linked together, a back and front and on different paths to the "zone"
